plugin.video.vrt.nu/resources/lib/vrtplayer/urltostreamservice.py

```diff
@@ -150,39 +185,54 @@ def _get_license_key(self, key_url, key_type='R', key_headers=None, key_value=No
             return ''.join((key_url, '|', header.strip('&'), '|', key_value, '|'))
 
 
-    def get_stream_from_url(self, url):
-        url = urljoin(self._vrt_base, url)
-        html_page = requests.get(url).text
+    def get_stream_from_url(self, url, xvrttoken=None):
```

Comment:
this method is becomming quite bulky cant we split it up in some more logical pieces? To improve readability and maintenance